api/modules/projects/controller.js

```diff
@@ -1,144 +1,158 @@
+const Publisher = require(`../../classes/publisher`);
 
-controller.update = function(req, reply) {
-  return BaseController.prototype.update.call(this, req, reply, dateTracker.convertToISOStrings(true));
-};
+const FilesModel = require(`../files/model`);
+const PublishedFilesModel = require(`../publishedFiles/model`);
 
-controller.publishProject = function(req, reply) {
-  var result = Promise.resolve().then(function() {
-    var record = req.pre.records.models[0];
+const ProjectsModel = require(`./model`);
 
-    return Publisher.publish(record)
-    .then(dateTracker.convertToISOStrings(true))
-    .then(function(publishedRecord) {
-      return req.generateResponse(publishedRecord).code(200);
-    });
-  })
-  .catch(errors.generateErrorResponse);
+const DateTracker = require(`../../../lib/utils`).DateTracker;
 
-  return reply(result);
-};
+class PublishedFilesCleanup {
+  constructor(project) {
+    this.project = project;
+  }
 
-controller.unpublishProject = function(req, reply) {
-  var result = Promise.resolve().then(function() {
-    var record = req.pre.records.models[0];
+  _fetchPublishedFiles(fileModel) {
+    return PublishedFilesModel.query({
+      where: {
+        file_id: fileModel.get(`id`)
+      }
+    })
+    .fetchAll();
+  }
 
-    if (!record.attributes.publish_url) {
-      throw Boom.notFound(null, {
-        debug: true,
-        error: 'This project was not published'
-      });
+  _deletePublishedFiles(publishedFilesModels) {
+    if (publishedFilesModels.length === 0) {
+      return Promise.resolve();
     }
 
-    return Publisher.unpublish(record)
-    .then(dateTracker.convertToISOStrings(true))
-    .then(function(unpublishedRecord) {
-      return req.generateResponse(unpublishedRecord).code(200);
+    return publishedFilesModels.mapThen(function(publishedFileModel) {
+      return publishedFileModel.destroy();
     });
-  })
-  .catch(errors.generateErrorResponse);
-
-  return reply(result);
-};
-
-controller.delete = function(req, reply) {
-  var self = this;
-  var project = req.pre.records.models[0];
-
-  function clearAllPublishedFiles() {
-    function fetchFiles() {
-      return Files.query({
-        where: {
-          project_id: project.get('id')
-        }
-      }).fetchAll();
-    }
+  }
 
-    function fetchPublishedFiles(file) {
-      return PublishedFiles.query({
-        where: {
-          file_id: file.get('id')
-        }
-      }).fetchAll();
+  cleanup() {
+    return FilesModel.query({
+      where: {
+        project_id: this.project.get(`id`)
+      }
+    })
+    .fetchAll()
+    .then(filesModels => {
+      return filesModels.mapThen(fileModel =>
+        this._fetchPublishedFiles(fileModel)
+        .then(this._deletePublishedFiles)
+      );
+    });
+  }
+}
+
+class ProjectsController extends BaseController {
+  constructor() {
+    super(ProjectsModel);
+  }
+
+  formatRequestData(request) {
+    const now = new Date();
+    const data = {
```

Comment:
we're referencing `payload` thrugh `request` a lot here. it'd be easier on the eyes if we stored a reference to `payload`